engine/table/src/main/java/io/deephaven/engine/table/impl/locations/local/FileKeyValuePartitionLayout.java

```diff
@@ -0,0 +1,151 @@
+//
```

Comment:
The motivation for this refactoring was move all the file related logic from KeyValuePartitionLayout to FileKeyValuePartitionLayout and then URIStreamKeyValuePartitionLayout can extend from KeyValuePartitionLayout.

I checked with Andy that KeyValuePartitionLayout is being used on enterprise side, so after this PR, they would need to do a rename from KeyValuePartitionLayout -> FileKeyValuePartitionLayout, which he is okay with. Once Ryan is okay with the new design, I will ping Andy and he said he can verify if their code works with this PR.

Reply:
Verified with Andy, the new change works fine for him.